ambry-replication/src/main/java/com/github/ambry/replication/ReplicaThread.java

```diff
@@ -715,12 +717,16 @@ void fixMissingStoreKeys(ConnectedChannel connectedChannel, List<RemoteReplicaIn
             + " should be the same and greater than zero");
       }
       DataNodeId remoteNode = replicasToReplicatePerNode.get(0).getReplicaId().getDataNodeId();
-      GetResponse getResponse =
+      getResponse =
           getMessagesForMissingKeys(connectedChannel, exchangeMetadataResponseList, replicasToReplicatePerNode,
               remoteNode, remoteColoGetRequestForStandby);
       writeMessagesToLocalStoreAndAdvanceTokens(exchangeMetadataResponseList, getResponse, replicasToReplicatePerNode,
           remoteNode, remoteColoGetRequestForStandby);
     } finally {
+      if (getResponse != null && getResponse.getInputStream() instanceof NettyByteBufDataInputStream) {
+        // if the InputStream is NettyByteBufDataInputStream based, it's time to release its buffer.
+        ((NettyByteBufDataInputStream) (getResponse.getInputStream())).getBuffer().release();
+      }
```

Comment:
this not going to work, since the underlying input stream for getResponse is a `DataInputStream` not a `NettyByteBufDataInputStream`. check out line 1072.

We can make ChannelOutput return `DataInputStream` in `getInputStream` method, this would fix the problem.